tests/timeline.py

```diff
@@ -0,0 +1,148 @@
+from __future__ import annotations
```

Comment:
we can probably get rid of this

Reply:
My impression is this is needed to use newer typing:

```
________________________________ ERROR collecting tests/samples.py _________________________________
import file mismatch:
imported module 'tests.samples' has this __file__ attribute:
  /home/allen/ics-py/tests/samples
which is not the same as the test file we want to collect:
  /home/allen/ics-py/tests/samples.py
HINT: remove __pycache__ / .pyc files and/or use a unique basename for your test file modules
________________________________ ERROR collecting tests/timeline.py ________________________________
tests/timeline.py:61: in <module>
    def test_on(calendar: Calendar, when: date | datetime, expected_events: list[str]) -> None:
E   TypeError: unsupported operand type(s) for |: 'type' and 'type'

```

Reply:
Oh, this is the first future import I see that might actually make sense in modern python. This enables the deferred evaluation of annotations, so python no longer tries to interpret this annotation at runtime and instead simply stores it as string:
https://docs.python.org/3/library/__future__.html
I guess in this specific case the issue might be that the pipe for type unions is python 3.10 syntax and you're using an older python?
https://peps.python.org/pep-0604/
Anyways, it should be fine if you leave it in.